packages/xml-views-validation/test/test-utils.ts

```diff
@@ -34,11 +34,18 @@ export function testValidationsScenario(opts: {
 
 export function computeExpectedRanges(markedXMLSnippet: string): OffsetRange[] {
   const expectedRanges = [];
+
+  function previousMarkersOffset(): number {
+    // "🢂".length === 2 (due to surrogate pairs in UTF-8)
+    // Two markers per previous range.
+    return expectedRanges.length * 2 * 2;
```

Comment:
Instead of magic numbers 2 and 2 you can use
```suggestion
    return expectedRanges.length * ("🢂".length + "🢀".length);
```

Reply:
will extract the marker signs to variables as well, it reduces readability (no more literal regExps), but makes it easy to update the marker arrows in the future.

Reply:
I'm not sure that's necessary, they will remain literal in tests

Reply:
And it's easy enough to find-and-replace these characters